acc_acc_modules/1_annotate_acc_modules.py

```diff
@@ -0,0 +1,373 @@
+
+# +
+# pao1_median_subset = subset_pao1_compendium.median().to_frame("median subset expression")
+# pa14_median_subset = subset_pa14_compendium.median().to_frame("median subset expression")
+# -
+
+# Add median expression to gene ids
+pao1_gene_annot = pao1_gene_module_labels.merge(
+    pao1_median_all, left_index=True, right_index=True, how="left"
+)
+pa14_gene_annot = pa14_gene_module_labels.merge(
+    pa14_median_all, left_index=True, right_index=True, how="left"
+)
+
+# Add median subset expression to gene ids
+"""pao1_gene_annot = pao1_gene_annot.merge(
+    pao1_median_subset, left_index=True, right_index=True, how="left"
+)
+pa14_gene_annot = pa14_gene_annot.merge(
+    pa14_median_subset, left_index=True, right_index=True, how="left"
+)"""
+
+print(pao1_gene_annot.shape)
+pao1_gene_annot.head()
+
+print(pa14_gene_annot.shape)
+pa14_gene_annot.head()
+
+
+# ## Genome location information
+#
+# How far are genes from other genes in the same module?
+
+# +
+# TO DO: Move into scripts
+# For genes in the same module, calculate the pairwise distance from each other
+# Calculate the median pairwise distance to represent how spread the module is
+# across the genome
+# Other metrics?
+
+
+def get_intra_module_dist(annot_df, pa_prefix):
+    rows = []
+    for grp_name, grp_df in annot_df.groupby("module id"):
+        # print("module", grp_name)
+
+        # Trim off "PA" and convert number to integer
+        ids = grp_df.index
+
+        # Convert trailing id numbers to floats
+        num_ids = [float(_id.split(pa_prefix)[1]) for _id in ids]
+
+        abs_dist = []
+        for gene1, gene2 in product(num_ids, num_ids):
+            if gene1 != gene2:
+                dist = abs(gene1 - gene2)
+                # print(gene1, gene2, dist)
+                abs_dist.append(dist)
+
+        median_module_dist = np.median(abs_dist)
+        min_dist = np.min(abs_dist)
+        max_dist = np.max(abs_dist)
+
+        for _id in ids:
+            rows.append(
+                {
+                    "gene id": _id,
+                    "median pairwise dist": median_module_dist,
+                    "min pairwise dist": min_dist,
+                    "max pairwise dist": max_dist,
+                }
+            )
+
+    module_dist = pd.DataFrame(rows)
+    module_dist = module_dist.set_index("gene id")
+
+    return module_dist
+
+
+# -
+
+pao1_module_dist = get_intra_module_dist(pao1_gene_annot, pa_prefix="PA")
+pa14_module_dist = get_intra_module_dist(pa14_gene_annot, pa_prefix="PA14_")
+
+pao1_module_dist.head(10)
+
+pa14_module_dist.head(10)
+
+# Add module distance to gene names
+pao1_gene_annot = pao1_gene_annot.merge(
+    pao1_module_dist, left_index=True, right_index=True, how="left"
+)
+pa14_gene_annot = pa14_gene_annot.merge(
+    pa14_module_dist, left_index=True, right_index=True, how="left"
+)
+
+# ## Add KEGG pathways
+#
+# For each pathway, what genes are contained in it. This information is only available for PAO1.
```

Comment:
Does this mean you have info for the core genes and the PA01 accessory genes? Wouldn't the core gene modules be the same across PA01 and PA14?

Reply:
Just realized this whole thing is only focusing on accessory genes. Nvm!